File: backend/schemas/menu.py
from pydantic import BaseModel


class DesjejumSchema(BaseModel):
    inicio: str = "07:00"
    fim: str = "08:00"
    bebida: list[str]
    pao: str
    raiz_ou_farinaceio: str
    fruta: str
    ovolactovegetariano: str


class AlmocoSchema(BaseModel):
    acompanhamento_I: str
    acompanhamento_II: str
    guarnicao: str
    salada_cozida: str
    salada_crua: str
    proteina: str
    opcao_proteina: str
    fruta: str
    ovolactovegetariano: list[str]
    suco: str


class JantarSchema(BaseModel):
    bebida: list[str]
    pao: str
    proteina: str
    raiz_ou_farinaceio: str
    sopa: str | None = None
    ovolactovegetariano: list[str]


class RefeicaoItem(BaseModel):
    desjejum: DesjejumSchema | None = None
    almoco: AlmocoSchema | None = None
    jantar: JantarSchema | None = None


class DiaCardapio(BaseModel):
    dia: str
    data: str
    refeicoes: list[RefeicaoItem]


class CardapioResponseSchema(BaseModel):
    data_inicio: str
    data_fim: str
    cardapio: list[DiaCardapio]

File: frontend/src/types/menu.ts
export interface RefeicaoDia {
  desjejum: {
    bebida: string[];
    pao: string;
    proteina: string;
    raiz_ou_farinaceio: string;
    fruta: string;
    ovolactovegetariano: string;
  };
  almoco: {
    acompanhamento_I: string;
    acompanhamento_II: string;
    guarnicao: string;
    salada_cozida: string;
    salada_crua: string;
    proteina: string;
    opcao_proteina: string;
    fruta: string;
    ovolactovegetariano: string[];
    suco: string;
  };
  jantar: {
    bebida: string[];
    pao: string;
    proteina: string;
    raiz_ou_farinaceio: string;
    sopa: string;
    ovolactovegetariano: string[];
  };
}

export interface DailyMenu {
  dia: string; // Substitui 'dayOfWeek'
  data: string; // Substitui 'date'
  refeicoes: RefeicaoDia[];
  created_at?: string;
}

export interface WeeklyMenuResponse {
  data_inicio: string;
  data_fim: string;
  cardapio: DailyMenu[];
  created_at?: string;
}

export type MealType = "breakfast" | "lunch" | "dinner";

export interface CurrentMealResponse {
  mealType: MealType;
  isActive: boolean;
}

export interface RestaurantStatus {
  isOpen: boolean;
  isLastServed: boolean;
  defaultMeal: MealType | string;
  badgeText: string;
}
